app/Game/game.cpp

```diff
@@ -63,71 +90,95 @@ int Game::run()
     Point startPos;
     startPos.x = 1;
     startPos.y = 1;
-    int dir = 1, len = 5;
-    initSnake(startPos, dir, len);
+    int dir = 1;
+    int len = 5;
+    if(!initSnake(startPos, dir, len))
+    {
+        return 1;
+    }
 
     /*
      * Initializing the Ball and generating it on the field
      */
-    Ball ball;
-    initBall(&ball);
-
+    labyrinth.initBall();
     /*
-     * displaying the starting state of labyrinth
+     * displaying the starting state of labyrinth(forcing to display it Full if possible)
      */
-    labyrinth.displayHandler();
+    labyrinth.displayHandler(DISPFULL);
 
     bool flag = false;
-    while (!flag)
+    while (!flag && cycles != 0)
     {
         int command = getch();
         switch (command)
         {
             case ERR:
+            {
                 break;
+            }
             case KEY_UP:
+            {
                 snake.setDirection(MVUP);
                 break;
+            }
             case KEY_DOWN:
+            {
                 snake.setDirection(MVDOWN);
                 break;
+            }
             case KEY_LEFT:
+            {
                 snake.setDirection(MVLEFT);
                 break;
+            }
             case KEY_RIGHT:
+            {
                 snake.setDirection(MVRIGHT);
                 break;
+            }
             case 'q':
+            {
                 flag = true;
                 break;
+            }
             case 's':
+            {
                 char fName[] = "testsave";
```

Comment:
What does this block of code do?

Reply:
On the press of the 's' key saves current labyrinth to a file by a name "testsave", displays message of if the writing to the file was successful , and freezes game for 2*"sleep time"
Then redraws the screen
Mainly it's left as an example of usage and can be removed

Reply:
Well, if it is not important I think this should be removed